strategy.go

```diff
@@ -0,0 +1,182 @@
+package kafka
+
+import "sort"
+
+// strategy encapsulates the client side rebalancing logic
+type strategy interface {
+	// ProtocolName of strategy
+	ProtocolName() string
```

Comment:
Any reason not to non-exported method names in this interface? (since the interface is non-exported itself)

Reply:
Two reasons, neither strong.  

The first is I have a personal pattern where I extend the Go style to apply to struct > struct communication.  Even if the type is lower case, I prefer the func to be uppercase to clearly indicate that it should be shared.

The second is that strategy is similar to Balancer, you can easily see others wanting to implement their own Strategies.  For example, a rack-aware consumer strategy.  In that case, uppercasing strategy is all that's required.

Reply:
Fair, I can see us exposing the strategy interface at some point :+1: